### DISCUSSION TITLE
Question about "self-signing identifiers" as alluded to in this FAQ

### DISCUSSION THREAD

**Discussion by vdods** [WebOfTrust/keri#43] [Q&A]
Hi there, I've been working with SAIDs (self-addressing identifiers -- a nice concept!), and a natural and obvious corollary of SAIDs would be "self-signing identifiers", in which instead of a hash derived from and embedded within the document, a digital signature is used (and the public key is specified somehow; probably within the document itself).

I did some searching and all I found was [this FAQ](https://identity.foundation/keri/docs/Q-and-A-Security.html):

    *Q: What do I need a self-signing identifier for?

    Often it is an efficiecy measure where the identifier includes the signature as your content-addressable hash.
    (SamMSmith)

So presumably this concept exists already.  My question is: is there a specification somewhere?  Is there any software implementing this concept?

Currently I'm using the `said` Rust crate within the github.com/THCLab/cesrox for SAID functionality, and I'm trying to figure out if there's an analog for "SSIDs", or if I should try to design/implement such a thing myself.

Thanks!

**Comment by kentbull** [comment]
To my understanding the concept of a [self certifying identifier](https://github.com/WebOfTrust/WOT-terms/wiki/self-certifying-identifier), SCID, is what you are looking for. See Section 2 of Dr. Smith's [KERI whitepaper](https://github.com/SmithSamuelM/Papers/blob/master/whitepapers/KERI_WP_2.x.web.pdf).

**Comment by SmithSamuelM** [comment]
A self-signing hash is a misnomer. Signing requires a key-pair by which one can prove control. In the KERI white paper the concept of a self-signing  AID (autonomic identifier) is introduced. It was never implemented because we couldn’t find a use case that anyone cared about. The problem with an identifier that is the signature on its inception event is that when you rotate keys, the signature that is the identifier is no longer self-signing. So its only useful for non-transferable identifiers (i.e. ephemeral identifiers whose keys can not be rotated). Given that signatures are normally twice the length of hashes or keys, its more verbose if the AID is referenced in multiple contexts. But in a narrow use case a self-signing aid (non-transferable i.e. non-rotatable) is more compact because no signature need be attached.  If someone has a valid use case, we could add a CESR code for just such an identifier.
